libcorsaro/libcorsaro_avro.c

```diff
@@ -212,10 +216,20 @@ int corsaro_read_next_avro_record(corsaro_avro_reader_t *reader,
 
 int corsaro_close_avro_writer(corsaro_avro_writer_t *writer) {
 
-    if (writer->out) {
-        avro_file_writer_close(writer->out);
+    if (writer->out == NULL) {
+        return 0;
     }
+
+    avro_file_writer_close(writer->out);
     writer->out = NULL;
+
+    char donebuf[1024];
+    if (snprintf(donebuf, sizeof(donebuf), "%s.done", writer->fname)
+        >= sizeof(donebuf)) {
+        corsaro_log(writer->logger, "unable to build .done file name");
+        return -1;
+    }
+    fclose(fopen(donebuf, "w"));
```

Comment:
In theory `fopen` could return NULL if there was an error, which won't play well as an argument to `fclose`.

Pretty rare occurrence and you've likely got bigger problems if this `fopen` fails, but you never know...